File: src/main/java/com/github/spartusch/webquery/WebQuery.java
package com.github.spartusch.webquery;

import java.nio.charset.Charset;
import java.util.Objects;

public class WebQuery {

    private byte[] content;
    private Charset charset;

    public WebQuery(final String content, final Charset charset) {
        this.content = content.getBytes(charset);
        this.charset = charset;
    }

    public byte[] getContent() {
        return content;
    }

    public Charset getCharset() {
        return charset;
    }

    public String getContentType() {
        return "text/plain; charset=" + charset.name();
    }

    public int getContentLength() {
        return content.length;
    }

    public String getContentDisposition(final String fileName) {
        Objects.requireNonNull(fileName);
        return "attachment; filename=" + fileName;
    }
}

File: src/main/kotlin/com/github/spartusch/webquery/WebQuery.kt
package com.github.spartusch.webquery

import java.nio.charset.Charset

class WebQuery internal constructor(content: String, val charset: Charset) {

    val contentBytes = content.toByteArray(charset)
    val contentLength get() = contentBytes.size
    val contentType get() = "text/plain; charset=${charset.name()}"

    override fun toString() = String(contentBytes, charset)

    companion object {
        @JvmStatic
        fun getContentDisposition(fileName: String) = "attachment; filename=$fileName"
    }

}
